### DISCUSSION TITLE
Guidance to run a reconstruction demo

### DISCUSSION THREAD

**Discussion by JIANG-CX** [gira3d/gira3d-reconstruction#4] [Q&A] [answered]
Wonderful work！
Could you please provide guidance on how to run a reconstruction demo similar to the one shown in the video? Thank you.

**Comment by kshitijgoel007** [comment]
Please take a look at this script https://github.com/gira3d/sogmm_py/blob/697cbeaf10c60fa80445c25e05dd9d720a296869/run_scripts/remote_isogmm.py

You have to modify it a bit to get it to run on a custom dataset and a custom machine.

**Comment by JIANG-CX** [comment]
ok, thanks.

获取 Outlook for iOS<https://aka.ms/o0ukef>
________________________________
发件人: Kshitij Goel ***@***.***>
发送时间: Thursday, June 20, 2024 1:13:12 AM
收件人: gira3d/gira3d-reconstruction ***@***.***>
抄送: JIANG Chenxing ***@***.***>; Author ***@***.***>
主题: Re: [gira3d/gira3d-reconstruction] Guidance to run a reconstruction demo (Issue #2)


Please take a look at this script https://github.com/gira3d/sogmm_py/blob/697cbeaf10c60fa80445c25e05dd9d720a296869/run_scripts/remote_isogmm.py

You have to modify it a bit to get it to run on a custom dataset and a custom machine.

―

**Comment by kshitijgoel007** [comment]
For visualization, you can reconstruct from the SOGMM and plot using Open3D. This function will be helpful https://github.com/gira3d/sogmm_py/blob/697cbeaf10c60fa80445c25e05dd9d720a296869/run_scripts/isogmm_analysis.py#L117

Open3D docs will also be helpful.

**Comment by JIANG-CX** [comment]
ok, I will try it. Thanks for your assistance.

获取 Outlook for iOS<https://aka.ms/o0ukef>
________________________________
发件人: Kshitij Goel ***@***.***>
发送时间: Thursday, June 20, 2024 1:14:49 AM
收件人: gira3d/gira3d-reconstruction ***@***.***>
抄送: JIANG Chenxing ***@***.***>; Author ***@***.***>
主题: Re: [gira3d/gira3d-reconstruction] Guidance to run a reconstruction demo (Issue #2)


For visualization, you can reconstruct from the SOGMM and plot using Open3D. This function will be helpful https://github.com/gira3d/sogmm_py/blob/697cbeaf10c60fa80445c25e05dd9d720a296869/run_scripts/isogmm_analysis.py#L117

Open3D docs will also be helpful.

―

**Comment by JIANG-CX** [comment]
Hi,
I set up the environment following the instructions in the README and ran the remote_isogmm.py script with the command:
`
python3 wet/src/sogmm_py/run_scripts/remote_isogmm.py
`
However, I encountered an error:
`
ModuleNotFoundError: No module named 'sogmm_py'
`
I then tried running python3 setup.py install in the directory:
`
gira3d-reconstruction/wet/src
`
But I received another error:
`
No module named 'sogmm_py.liegroups'
`
Could you please help me resolve this issue? Thanks.

**Comment by kshitijgoel007** [comment]
In the root directory of this colcon package, run the workon script https://github.com/gira3d/gira3d-reconstruction/blob/master/workon

Using
```bash
source workon
```

**Comment by kshitijgoel007** [comment, marked as answer]
Also the official installation instructions do not require you to run `python3 setup.py install`. Please follow the instructions here: https://github.com/gira3d/gira3d-reconstruction/blob/master/README.md#build-and-install

**Comment by JIANG-CX** [comment]
Thanks for your reply. The setup problem has been solved, thanks. ^_^

获取 Outlook for iOS<https://aka.ms/o0ukef>
________________________________
发件人: Kshitij Goel ***@***.***>
发送时间: Friday, June 21, 2024 2:21:48 AM
收件人: gira3d/gira3d-reconstruction ***@***.***>
抄送: JIANG Chenxing ***@***.***>; Author ***@***.***>
主题: Re: [gira3d/gira3d-reconstruction] Guidance to run a reconstruction demo (Issue #2)


Also the official installation instructions do not require you to run python3 setup.py install. Please follow the instructions here: https://github.com/gira3d/gira3d-reconstruction/blob/master/README.md#build-and-install

―

**Comment by kshitijgoel007** [comment]
You are welcome. Please do not hesitate to ask further questions through the discussions portal: https://github.com/gira3d/gira3d-reconstruction/discussions
